android/app/src/main/java/com/criticalblue/auth/demo/auth/AuthRepo.java

```diff
@@ -61,45 +63,92 @@
 
     private AuthLoginListener loginListener;
-    private AuthLogoutListener logoutListener;
 
-    private AuthorizationService authService;
+    private final AuthorizationService authService;
     private AuthState authState;
     private String userInfoUrl;
+
     String clientId;
     String redirectUri;
     String authScope;
 
-    ConnectionBuilder connectionBuilder;
-
     public AuthRepo(BooksApp app) {
         this.app = app;
 
         loginLock = new Semaphore(1);
 
         loginListener = null;
-        logoutListener = null;
-
-        connectionBuilder = new SignedConnectionBuilder(app.getAssets());
-
-        AppAuthConfiguration.Builder builder = new AppAuthConfiguration.Builder();
-        builder.setConnectionBuilder(connectionBuilder);
-        authService = new AuthorizationService(app, builder.build());
         authState = null;
         userInfoUrl = null;
         clientId = null;
         redirectUri = null;
         authScope = null;
+
+        authService = new AuthorizationService(app, ApproovCustomConnection());
+    }
+
+    /**
+     * All the calls made by AppAuth will be secured by using the Approov Managed Trust Roots
+     * feature to protect the TLS channel against MitM attacks.
+     *
+     * @link https://approov.io/docs/latest/approov-usage-documentation/#managed-trust-roots
+     *
+     * @return The AppAuthConfiguration with a HttpsURLConnection secured by Approov
+     */
+    private AppAuthConfiguration ApproovCustomConnection() {
+        return new AppAuthConfiguration.Builder()
+                .setConnectionBuilder(new ConnectionBuilder() {
+                    @NonNull
+                    @Override
+                    public HttpsURLConnection openConnection(@NonNull Uri uri) throws IOException {
+                        URL url = new URL(uri.toString());
+                        HttpsURLConnection connection = (HttpsURLConnection) url.openConnection();
+
+                        io.approov.service.httpsurlconn.ApproovService.addApproov(connection);
+
+                        return connection;
+                    }
+                })
+                .build();
     }
 
+    /**
+     * Approov allows to fetch secrets just in time for being used to make API calls, provided that
+     * the mobile app passes the attestation, thus no more hardcoded secrets on the app code.
+     *
+     * @link https://approov.io/docs/latest/approov-usage-documentation/#secure-strings
+     *
+     * @return The client secret retrieved from the Approov Cloud service
+     */
+    private String fetchJustInTimeClientSecret() {
+        String clientSecret = null;
+
+        try {
+            clientSecret = io.approov.service.httpsurlconn.ApproovService.fetchSecureString("client_secret", null);
+        } catch (ApproovRejectionException e) {
+            // failure due to the attestation being rejected, e.getARC() and e.getRejectionReasons() may be used to present information to the user
```

Comment:
These comments for all the catches are just the generic ones and could be removed

Reply:
I kept the info in the comments to help devs understanding their options, but I ma happy to remove them.